From 86b6d507806319da6fddac4237bb322ed3eaab95 Mon Sep 17 00:00:00 2001
From: c4pp4
Date: Tue, 13 Oct 2026 14:18:09 +0200
Subject: [PATCH 1/1] Fix qt6 compilation

Qt 6 defines Q_DECL_EXPORT as protected visibility, which breaks
linking of the bundled runner and tests against the library.

Signed-off-by: c4pp4
---
 src/libqtdbustest/DBusService.h         | 2 +-
 src/libqtdbustest/DBusTestRunner.h      | 2 +-
 src/libqtdbustest/QProcessDBusService.h | 2 +-
 src/libqtdbustest/SuicidalProcess.h     | 2 +-
 src/qdbus-test-runner/CMakeLists.txt    | 5 +++++
 5 files changed, 9 insertions(+), 4 deletions(-)

diff --git a/src/libqtdbustest/DBusService.h b/src/libqtdbustest/DBusService.h
--- a/src/libqtdbustest/DBusService.h
+++ b/src/libqtdbustest/DBusService.h
@@ -30,7 +30,7 @@
 
 typedef QSharedPointer<DBusService> DBusServicePtr;
 
-class Q_DECL_EXPORT DBusService {
+class __attribute__((visibility("default"))) DBusService {
 public:
 	DBusService(const QString &name, QDBusConnection::BusType busType);
 
diff --git a/src/libqtdbustest/DBusTestRunner.h b/src/libqtdbustest/DBusTestRunner.h
--- a/src/libqtdbustest/DBusTestRunner.h
+++ b/src/libqtdbustest/DBusTestRunner.h
@@ -29,7 +29,7 @@
 
 class DBusTestRunnerPrivate;
 
-class Q_DECL_EXPORT DBusTestRunner {
+class __attribute__((visibility("default"))) DBusTestRunner {
 public:
 	explicit DBusTestRunner(const QString &dbusSessionConfigFile =
 			DBUS_SESSION_CONFIG_FILE, const QString &dbusSystemConfigFile =
diff --git a/src/libqtdbustest/QProcessDBusService.h b/src/libqtdbustest/QProcessDBusService.h
--- a/src/libqtdbustest/QProcessDBusService.h
+++ b/src/libqtdbustest/QProcessDBusService.h
@@ -29,7 +29,7 @@
 
 class QProcessDBusServicePrivate;
 
-class Q_DECL_EXPORT QProcessDBusService: public DBusService {
+class __attribute__((visibility("default"))) QProcessDBusService: public DBusService {
 
 public:
 	explicit QProcessDBusService(const QString &name,
diff --git a/src/libqtdbustest/SuicidalProcess.h b/src/libqtdbustest/SuicidalProcess.h
--- a/src/libqtdbustest/SuicidalProcess.h
+++ b/src/libqtdbustest/SuicidalProcess.h
@@ -26,7 +26,7 @@
 
 class SuicidalProcessPrivate;
 
-class Q_DECL_EXPORT SuicidalProcess: public QProcess {
+class __attribute__((visibility("default"))) SuicidalProcess: public QProcess {
 Q_OBJECT
 public:
 	explicit SuicidalProcess(QObject *parent = 0);
diff --git a/src/qdbus-test-runner/CMakeLists.txt b/src/qdbus-test-runner/CMakeLists.txt
--- a/src/qdbus-test-runner/CMakeLists.txt
+++ b/src/qdbus-test-runner/CMakeLists.txt
@@ -4,6 +4,11 @@
 	qdbus-simple-test-runner.cpp
 )
 
+set_target_properties(
+	qdbus-simple-test-runner
+	PROPERTIES OUTPUT_NAME "qdbus-simple-test-runner${QT_SUFFIX}"
+)
+
 target_link_libraries(
 	qdbus-simple-test-runner
 	qtdbustest${QT_SUFFIX}
-- 
2.52.0
